packages/jetpack-ai-calypso/src/logo-generator/components/generator-modal.tsx

```diff
@@ -90,9 +92,22 @@ export const GeneratorModal: React.FC< GeneratorModalProps > = ( {
 		// First fetch the feature data so we have the most up-to-date info from the backend.
 		try {
 			const feature = await getFeature();
+			const hasHistory = ! isLogoHistoryEmpty( String( siteId ) );
+			const logoCost = feature?.costs?.[ 'jetpack-ai-logo-generator' ]?.logo ?? DEFAULT_LOGO_COST;
+			const promptCreationCost = 1;
+			const currentLimit = feature?.currentTier?.value || 0;
+			const currentUsage = feature?.usagePeriod?.requestsCount || 0;
+			const isUnlimited = currentLimit === 1;
+
+			// The user needs an upgrade immediately if they have no logos and not enough requests remaining for one prompt and one logo generation.
+			const needsMoreRequests =
+				! isUnlimited &&
+				! hasHistory &&
+				currentLimit - currentUsage < logoCost + promptCreationCost;
```

Comment:
We may need an extra check for the 1000 requests plan, since it's soft limited: we could have `currentLimit - currentUsage` as a negative number, and still be allowed to do the requests. Maybe an special check as the one did for the unlimited plan will sufice.